src/ds/d500/d500-options.cpp

```diff
@@ -50,4 +50,33 @@ namespace librealsense
 
         return temperature;
     }
+    temperature_xu_option::temperature_xu_option(const std::weak_ptr<uvc_sensor>& ep, 
+        platform::extension_unit xu, uint8_t id, 
+        std::string description, bool allow_set_while_streaming)
+        : uvc_xu_option<int16_t>(ep, xu, id, description, allow_set_while_streaming)
+    {
+        // defining the parsing modifier, to be used on the calls for query method
+        _parsing_modifier = [](const int16_t read_value) {
```

Comment:
This looks like magic to me.
IMO `_parsing_modifier` should be received in the Ctor of `uvc_xu_option` def value or we should have a special `uvc_xu_custom_option`